mealie/core/security/security.py

```diff
@@ -128,10 +129,35 @@ def user_from_ldap(db: AllRepositories, username: str, password: str) -> Private
     return user
 
 
-def authenticate_user(session, email: str, password: str) -> PrivateUser | bool:
+def _create_new_jwt_user(db: AllRepositories, claims: dict) -> PrivateUser:
+    settings = get_app_settings()
+    return db.users.create(
+        {
+            "full_name": claims[settings.JWT_AUTH_NAME_CLAIM],
+            "username": claims[settings.JWT_AUTH_USERNAME_CLAIM],
+            "email": claims[settings.JWT_AUTH_EMAIL_CLAIM],
+            "password": hash_password(secrets.token_urlsafe(13)),  # 13 char long random password
+            "group": settings.DEFAULT_GROUP,
+            "admin": True,
+        }
+    )
+
+
+def authenticate_user(session, email: str, password: str, jwt_assertion: str | None = None) -> PrivateUser | bool:
     settings = get_app_settings()
 
     db = get_repositories(session)
+
+    if settings.JWT_AUTH_ENABLED and jwt_assertion is not None:
+        try:
+            jwt_claims = get_claims_from_jwt_assertion(jwt_assertion)
+        except Exception:
+            return False
+        user = db.users.get_one(jwt_claims[settings.JWT_AUTH_EMAIL_CLAIM], "email", any_case=True)
+        if user is None and settings.JWT_AUTH_AUTO_SIGN_UP:
+            user = _create_new_jwt_user(db, jwt_claims)
+        return user
```

Comment:
This is the most important piece of code in the PR. When doing the user authentication, if `JWT_AUTH_ENABLED` is set to True and the JWT assertion is present, the JWT Auth flow will kick in and return a user (or an error), instead of letting the normal authentication user run.

The  user information will be retrieved from the decoded JWT, and this user will be returned directly if the process succeeds. There is the possibility to create users on the fly or return an error based on the value set in `JWT_AUTH_AUTO_SIGN_UP`